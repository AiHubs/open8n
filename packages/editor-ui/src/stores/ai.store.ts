import { defineStore } from 'pinia';
import * as aiApi from '@/api/ai';
import type { DebugErrorPayload } from '@/api/ai';
import { useRootStore } from '@/stores/n8nRoot.store';
import { useSettingsStore } from '@/stores/settings.store';
import { computed, ref } from 'vue';

export const useAIStore = defineStore('ai', () => {
	const rootStore = useRootStore();
	const settingsStore = useSettingsStore();

	const assistantChatOpen = ref(false);

	const isErrorDebuggingEnabled = computed(() => settingsStore.settings.ai.errorDebugging);

	async function debugError(payload: DebugErrorPayload) {
		return await aiApi.debugError(rootStore.getRestApiContext, payload);
	}

	return { isErrorDebuggingEnabled, debugError, assistantChatOpen };
});
