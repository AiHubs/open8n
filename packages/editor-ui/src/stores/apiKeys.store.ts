import { STORES } from '@/constants';
import type { ApiKey } from '@/Interface';
import { defineStore } from 'pinia';
import { useRootStore } from '@/stores/root.store';

import * as publicApiApi from '@/api/api-keys';
import { computed, ref } from 'vue';

export const useApiKeysStore = defineStore(STORES.API_KEYS, () => {
	const apiKeys = ref<ApiKey[]>([]);

	const rootStore = useRootStore();

	const apiKeysSortByCreationDate = computed(() =>
		apiKeys.value.sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()),
	);

	const apiKeysById = computed(() => {
		return apiKeys.value.reduce(
			(acc, apiKey) => {
				acc[apiKey.id] = apiKey;
				return acc;
			},
			{} as Record<string, ApiKey>,
		);
	});

	const getAllApiKeys = async () => {
		if (apiKeys.value.length) return apiKeys.value;
		apiKeys.value = await publicApiApi.getApiKeys(rootStore.restApiContext);
		return apiKeys.value;
	};

	const createApiKey = async (label: string) => {
		const newApiKey = await publicApiApi.createApiKey(rootStore.restApiContext, { label });
		apiKeys.value.push(newApiKey);
		return newApiKey;
	};

	const deleteApiKey = async (id: string) => {
		await publicApiApi.deleteApiKey(rootStore.restApiContext, id);
		apiKeys.value = apiKeys.value.filter((apiKey) => apiKey.id !== id);
	};

	const updateApiKey = async (id: string, data: { label: string }) => {
		await publicApiApi.updateApiKey(rootStore.restApiContext, id, data);
	};

	return {
		getAllApiKeys,
		createApiKey,
		deleteApiKey,
		updateApiKey,
		apiKeysSortByCreationDate,
		apiKeysById,
		apiKeys,
	};
});
