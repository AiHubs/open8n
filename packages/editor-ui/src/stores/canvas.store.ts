import { computed, ref } from 'vue';
import { defineStore } from 'pinia';
import { useWorkflowsStore } from '@/stores/workflows.store';
import type { INodeUi, XYPosition } from '@/Interface';
import { useLoadingService } from '@/composables/useLoadingService';

/**
 * @TODO Migrate loading service to ui store
 * @TODO Migrate panelHeight to ui store
 * @TODO Migrate aiNodes computed to ai store
 * @TODO Remove useCanvasStore
 */
export const useCanvasStore = defineStore('canvas', () => {
	const workflowStore = useWorkflowsStore();
	const loadingService = useLoadingService();

	const newNodeInsertPosition = ref<XYPosition | null>(null);
	const panelHeight = ref(0);

	const nodes = computed<INodeUi[]>(() => workflowStore.allNodes);
	const aiNodes = computed<INodeUi[]>(() =>
		nodes.value.filter((node) => node.type.includes('langchain')),
	);

	function setPanelHeight(height: number) {
		panelHeight.value = height;
	}

	return {
		newNodeInsertPosition,
		isLoading: loadingService.isLoading,
		aiNodes,
		panelHeight: computed(() => panelHeight.value),
		setPanelHeight,
		startLoading: loadingService.startLoading,
		setLoadingText: loadingService.setLoadingText,
		stopLoading: loadingService.stopLoading,
	};
});
