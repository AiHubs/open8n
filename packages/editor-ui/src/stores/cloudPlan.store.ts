import { computed, reactive } from 'vue';
import { defineStore } from 'pinia';
import type { CloudPlanState } from '@/Interface';
import { useRootStore } from '@/stores/n8nRoot.store';
import { useSettingsStore } from '@/stores/settings.store';
import { useUsersStore } from '@/stores/users.store';
import { getCurrentPlan, getCurrentUsage } from '@/api/cloudPlans';
import { DateTime } from 'luxon';

const DEFAULT_STATE: CloudPlanState = {
	data: null,
	usage: null,
};

export const useCloudPlanStore = defineStore('cloudPlan', () => {
	const rootStore = useRootStore();
	const settingsStore = useSettingsStore();
	const usersStore = useUsersStore();

	const state = reactive<CloudPlanState>(DEFAULT_STATE);

	const setData = (data: CloudPlanState['data']) => {
		state.data = data;
	};

	const setUsage = (data: CloudPlanState['usage']) => {
		state.usage = data;
	};

	const userIsTrialing = computed(() => state.data?.metadata?.group === 'trial');

	const currentPlanData = computed(() => state.data);

	const currentUsageData = computed(() => state.usage);

	const trialExpired = computed(
		() =>
			state.data?.metadata?.group === 'trial' &&
			DateTime.now().toMillis() >= DateTime.fromISO(state.data?.expirationDate).toMillis(),
	);

	const allExecutionsUsed = computed(() => {
		if (!state.usage?.executions || !state.data?.monthlyExecutionsLimit) return false;
		return state.usage?.executions >= state.data?.monthlyExecutionsLimit;
	});

	const getOwnerCurrentPLan = async () => {
		const cloudUserId = settingsStore.settings.n8nMetadata?.userId;
		const hasCloudPlan =
			usersStore.currentUser?.isOwner && settingsStore.isCloudDeployment && cloudUserId;
		if (!hasCloudPlan) throw new Error('User does not have a cloud plan');
		return getCurrentPlan(rootStore.getRestCloudApiContext, cloudUserId as string);
	};

	const getInstanceCurrentUsage = async () => {
		return getCurrentUsage({ baseUrl: rootStore.getBaseUrl, sessionId: '' });
	};

	return {
		setData,
		setUsage,
		getOwnerCurrentPLan,
		getInstanceCurrentUsage,
		userIsTrialing,
		currentPlanData,
		currentUsageData,
		trialExpired,
		allExecutionsUsed,
	};
});
