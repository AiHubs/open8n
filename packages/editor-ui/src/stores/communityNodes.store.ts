import * as communityNodesApi from '@/api/communityNodes';
import { getAvailableCommunityPackageCount } from '@/api/settings';
import { defineStore } from 'pinia';
import { useRootStore } from './n8nRoot.store';
import type { PublicInstalledPackage } from 'n8n-workflow';
import type { CommunityNodesState, CommunityPackageMap } from '@/Interface';
import { STORES } from '@/constants';
import { computed, ref } from 'vue';

const LOADER_DELAY = 300;

export const useCommunityNodesStore = defineStore(STORES.COMMUNITY_NODES, () => {
	const state = ref<CommunityNodesState>({ availablePackageCount: -1, installedPackages: {} });

	// ---------------------------------------------------------------------------
	// #region Computed
	// ---------------------------------------------------------------------------

	const getInstalledPackages = computed(() => {
		return Object.values(state.value.installedPackages).sort((a, b) =>
			a.packageName.localeCompare(b.packageName),
		);
	});

	// #endregion

	// ---------------------------------------------------------------------------
	// #region Methods
	// ---------------------------------------------------------------------------

	const fetchAvailableCommunityPackageCount = async (): Promise<void> => {
		if (state.value.availablePackageCount === -1) {
			state.value.availablePackageCount = await getAvailableCommunityPackageCount();
		}
	};

	const setInstalledPackages = (packages: PublicInstalledPackage[]) => {
		state.value.installedPackages = packages.reduce(
			(packageMap: CommunityPackageMap, pack: PublicInstalledPackage) => {
				packageMap[pack.packageName] = pack;
				return packageMap;
			},
			{},
		);
	};

	const fetchInstalledPackages = async (): Promise<void> => {
		const rootStore = useRootStore();
		const installedPackages = await communityNodesApi.getInstalledCommunityNodes(
			rootStore.getRestApiContext,
		);
		setInstalledPackages(installedPackages);
		const timeout = installedPackages.length > 0 ? 0 : LOADER_DELAY;
		setTimeout(() => {
			return;
		}, timeout);
	};

	const installPackage = async (packageName: string): Promise<void> => {
		try {
			const rootStore = useRootStore();
			await communityNodesApi.installNewPackage(rootStore.getRestApiContext, packageName);
			await fetchInstalledPackages();
		} catch (error) {
			throw error;
		}
	};

	const uninstallPackage = async (packageName: string): Promise<void> => {
		try {
			const rootStore = useRootStore();
			await communityNodesApi.uninstallPackage(rootStore.getRestApiContext, packageName);
			removePackageByName(packageName);
		} catch (error) {
			throw error;
		}
	};

	const removePackageByName = (name: string): void => {
		const { [name]: removedPackage, ...remainingPackages } = state.value.installedPackages;
		state.value.installedPackages = remainingPackages;
	};

	const updatePackageObject = (newPackage: PublicInstalledPackage) => {
		state.value.installedPackages[newPackage.packageName] = newPackage;
	};

	const updatePackage = async (packageName: string): Promise<void> => {
		try {
			const rootStore = useRootStore();
			const packageToUpdate: PublicInstalledPackage = getInstalledPackageByName.value(packageName);
			const updatedPackage: PublicInstalledPackage = await communityNodesApi.updatePackage(
				rootStore.getRestApiContext,
				packageToUpdate.packageName,
			);
			updatePackageObject(updatedPackage);
		} catch (error) {
			throw error;
		}
	};

	// #endregion

	const getInstalledPackageByName = computed(() => {
		return (name: string): PublicInstalledPackage => state.value.installedPackages[name];
	});

	const getAvailablePackageCount = computed(() => state.value.availablePackageCount);

	return {
		getInstalledPackageByName,
		getInstalledPackages,
		getAvailablePackageCount,
		fetchAvailableCommunityPackageCount,
		fetchInstalledPackages,
		installPackage,
		uninstallPackage,
		updatePackage,
	};
});
