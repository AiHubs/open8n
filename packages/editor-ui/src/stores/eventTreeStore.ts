import { defineStore } from "pinia";
import { AbstractMessageEventBusDestination } from "../components/SettingsLogStreaming/types";

export class EventNamesTreeCollection {
	label = '';
	_selected? = false;
	_indeterminate? = false;
	_name = '';
	children: EventNamesTreeCollection[] = [];
}

export interface TreeAndSelectionStoreItem {
		destination: AbstractMessageEventBusDestination,
		tree: EventNamesTreeCollection,
		selectedEvents: Set<string>,
		selectedLevels: Set<string>,
}

export interface TreeAndSelectionStore {
	[key:string]: TreeAndSelectionStoreItem
}

export const useEventTreeStore = defineStore('eventTree', {
  state: () => ({
		items: {} as TreeAndSelectionStore,
		eventNames: new Set<string>(),
		eventLevels: new Set<string>(),
	}),
  getters: {
  },
  actions: {
		addDestination(destination: AbstractMessageEventBusDestination) {
			if (destination.id in this.items) {
				this.items[destination.id].destination = destination;
			} else {
				this.setSelectionAndBuildItems(destination);
			}
		},
		getDestination(destinationId: string): AbstractMessageEventBusDestination | undefined {
			if (destinationId in this.items) {
				return this.items[destinationId].destination;
			} else {
				return;
			}
		},
		updateDestination(destination: AbstractMessageEventBusDestination) {
			if (destination.id in this.items) {
				this.items[destination.id].destination = destination;
			}
		},
		removeDestination(destinationId: string) {
			delete this.items[destinationId];
			if (destinationId in this.items) {
				this.$patch({items: {
					...this.items,
				}});
			}
		},
		clearDestinations() {
			this.items = {};
		},
		addEventName(name: string) {
			this.eventNames.add(name);
		},
		removeEventName(name: string) {
			this.eventNames.delete(name);
		},
		clearEventNames() {
			this.eventNames.clear();
		},
		addEventLevel(name: string) {
			this.eventLevels.add(name);
		},
		removeEventLevel(name: string) {
			this.eventLevels.delete(name);
		},
		clearEventLevels() {
			this.eventLevels.clear();
		},
		addSelectedEvent(id:string, name: string) {
			this.items[id]?.selectedEvents?.add(name);
			this.setSelectedInTree(id, name, true);
		},
		removeSelectedEvent(id:string, name: string) {
			this.items[id]?.selectedEvents?.delete(name);
			this.setSelectedInTree(id, name, false);
		},
		setSelectedInTree(id: string, name: string, isSelected: boolean) {
			const parts = name.split('.');
			let part: string | undefined;
			let children = this.items[id].tree.children;
			while ((part = parts.shift())) {
				if (part) {
					const foundChild = children.find((e) => e.label === part);
					if (foundChild) {
						if (parts.length === 0) {
							foundChild._selected = isSelected;
						}
						children = foundChild.children;
					} else {
						break;
					}
				}
			}
		},
		setIndeterminateInTree(id: string, name: string, isIndeterminate: boolean) {
			const parts = name.split('.');
			let part: string | undefined;
			let children = this.items[id].tree.children;
			while ((part = parts.shift())) {
				if (part) {
					const foundChild = children.find((e) => e.label === part);
					if (foundChild) {
						if (parts.length === 0) {
							foundChild._indeterminate = isIndeterminate;
						}
						children = foundChild.children;
					} else {
						break;
					}
				}
			}
		},
		addSelectedLevel(id:string, name: string) {
			this.items[id]?.selectedLevels?.add(name);
		},
		removeSelectedLevel(id:string, name: string) {
			this.items[id]?.selectedLevels?.delete(name);
		},
		setSelectedLevels(id:string, levelCheckList: string[]) {
			if (id in this.items) {
				this.items[id].selectedLevels = new Set<string>(levelCheckList);
			}
		},
		getSelectedLevels(id:string): string[] {
			return Array.from(this.items[id]?.selectedLevels.values()) ?? [];
		},
		getEventTree(id:string): EventNamesTreeCollection {
			return this.items[id]?.tree ?? {};
		},
		removeDestinationItemTree(id: string) {
			delete this.items[id];
		},
		clearDestinationItemTrees() {
			this.items = {} as TreeAndSelectionStore;
		},
		setSelectionAndBuildItems(destination: AbstractMessageEventBusDestination) {
			if (!(destination.id in this.items)) {
				this.items[destination.id] = {
					destination,
					tree: new EventNamesTreeCollection(),
					selectedEvents: new Set<string>(),
					selectedLevels: new Set<string>(),
				} as TreeAndSelectionStoreItem;
			}
			this.items[destination.id]?.selectedEvents?.clear();
			for (const eventName of destination.subscribedEvents) {
				this.items[destination.id]?.selectedEvents?.add(eventName);
			}
			for (const eventLevel of destination.subscribedLevels) {
				this.items[destination.id]?.selectedLevels?.add(eventLevel);
			}
			this.items[destination.id].tree = treeCollectionFromStringList(this.eventNames, this.items[destination.id]?.selectedEvents);
			// this.items[destination.id].elTree = elTreeFromStringList(this.eventNames);
		},
	},
});

export function treeCollectionFromStringList(dottedList: Set<string>, selectionList: Set<string> = new Set()) {
	const conversionResult = new EventNamesTreeCollection();
	dottedList.forEach((dottedString: string) => {
		const parts = dottedString.split('.');

		let part: string | undefined;
		let children = conversionResult.children;
		let partialName = '';
		while ((part = parts.shift())) {
			if (part) {
				const foundChild = children.find((e) => e.label === part);
				partialName += part;
				if (foundChild) {
					children = foundChild.children;
				} else {
					let _indeterminate = false;
					selectionList.forEach((e)=>{if (e.startsWith(partialName)) _indeterminate = true;});
					const newChild: EventNamesTreeCollection = {
						label: part,
						_name: partialName,
						_indeterminate,
						_selected: selectionList.has(partialName),
						children: [],
					};
					children.push(newChild);
					children = newChild.children;
				}
			}
			partialName += '.';
		}
	});
	return conversionResult;
};
