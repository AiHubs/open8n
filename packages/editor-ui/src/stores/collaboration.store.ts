import { defineStore } from 'pinia';
import { computed, ref } from 'vue';
import type { ActiveUsersForWorkflows, PushDataUsersForWorkflow } from '@/Interface';
import { usePushConnectionStore, useWorkflowsStore } from '@/stores';
import { STORES } from '@/constants';

export const useCollaborationStore = defineStore(STORES.COLLABORATION, () => {
	const pushStore = usePushConnectionStore();
	const workflowStore = useWorkflowsStore();

	const usersForWorkflows = ref<ActiveUsersForWorkflows>({});

	pushStore.addEventListener((event) => {
		if (event.type === 'activeWorkflowUsersChanged') {
			console.log('activeWorkflowUsersChanged', event.data);
			const activeWorkflowId = workflowStore.workflowId;
			if (event.data.workflowId === activeWorkflowId) {
				usersForWorkflows.value[activeWorkflowId] = event.data.activeUsers;
			}
		}
	});

	const workflowUsersUpdated = (data: ActiveUsersForWorkflows) => {
		usersForWorkflows.value = data;
	};

	const notifyWorkflowOpened = (workflowId: string) => {
		pushStore.send({
			type: 'workflowOpened',
			workflowId,
		});
	};

	const notifyWorkflowClosed = (workflowId: string) => {
		pushStore.send({ type: 'workflowClosed', workflowId });
	};

	const getUsersForCurrentWorkflow = computed(() => {
		return usersForWorkflows.value[workflowStore.workflowId];
	});

	return {
		usersForWorkflows,
		notifyWorkflowOpened,
		notifyWorkflowClosed,
		workflowUsersUpdated,
		getUsersForCurrentWorkflow,
	};
});
