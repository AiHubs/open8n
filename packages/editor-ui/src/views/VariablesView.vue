<script lang="ts" setup>
import { computed, ref } from 'vue';
import { useEnvironmentsStore, useUIStore, useSettingsStore } from '@/stores';
import { useI18n, useTelemetry, useToast, useUpgradeLink, useMessage } from '@/composables';

import ResourcesListLayout from '@/components/layouts/ResourcesListLayout.vue';
import VariablesRow from '@/components/VariablesRow.vue';

import { EnterpriseEditionFeature } from '@/constants';
import { DatatableColumn, EnvironmentVariable, TemporaryEnvironmentVariable } from '@/Interface';
import { uid } from 'n8n-design-system/utils';

const settingsStore = useSettingsStore();
const environmentsStore = useEnvironmentsStore();
const uiStore = useUIStore();
const telemetry = useTelemetry();
const i18n = useI18n();
const message = useMessage();

const { showError } = useToast();

const TEMPORARY_VARIABLE_UID_BASE = '@tmpvar';

const allVariables = ref<(EnvironmentVariable | TemporaryEnvironmentVariable)[]>([]);
const editMode = ref<Record<string, boolean>>({});

const datatableColumns = ref<DatatableColumn[]>([
	{
		id: 0,
		path: 'name',
		label: 'Name',
	},
	{
		id: 1,
		path: 'value',
		label: 'Value',
	},
	{
		id: 2,
		path: 'actions',
		label: '',
	},
]);

const isFeatureEnabled = computed(() =>
	settingsStore.isEnterpriseFeatureEnabled(EnterpriseEditionFeature.Variables),
);
const contextBasedTranslationKeys = computed(() => uiStore.contextBasedTranslationKeys);
const { upgradeLinkUrl } = useUpgradeLink({
	default: '&source=variables',
	desktop: '&utm_campaign=upgrade-variables',
});

async function initialize() {
	await environmentsStore.fetchAllVariables();

	allVariables.value = [...environmentsStore.variables];
}

function addTemporaryVariable() {
	const temporaryVariable: TemporaryEnvironmentVariable = {
		id: uid(TEMPORARY_VARIABLE_UID_BASE),
		key: '',
		value: '',
	};

	allVariables.value.unshift(temporaryVariable);
	editMode.value[temporaryVariable.id] = true;

	telemetry.track('User clicked add variable button');
}

async function saveVariable(data: EnvironmentVariable | TemporaryEnvironmentVariable) {
	let updatedVariable: EnvironmentVariable;

	try {
		if (typeof data.id === 'string' && data.id.startsWith(TEMPORARY_VARIABLE_UID_BASE)) {
			const { id, ...rest } = data;
			updatedVariable = await environmentsStore.createVariable(rest);
			allVariables.value.unshift(updatedVariable);
			allVariables.value = allVariables.value.filter((variable) => variable.id !== data.id);
		} else {
			updatedVariable = await environmentsStore.updateVariable(data as EnvironmentVariable);
			allVariables.value = allVariables.value.map((variable) =>
				variable.id === data.id ? updatedVariable : variable,
			);
			toggleEditingVariable(updatedVariable);
		}
	} catch (error) {
		showError(error, i18n.baseText('variables.errors.save'));
	}
}

function toggleEditingVariable(data: EnvironmentVariable) {
	editMode.value = {
		...editMode.value,
		[data.id]: !editMode.value[data.id],
	};
}

async function deleteVariable(data: EnvironmentVariable) {
	const confirmed = await message.confirm(
		i18n.baseText('variables.modals.deleteConfirm.message'),
		i18n.baseText('variables.modals.deleteConfirm.title'),
		{
			confirmButtonText: i18n.baseText('variables.modals.deleteConfirm.confirmButton'),
			cancelButtonText: i18n.baseText('variables.modals.deleteConfirm.cancelButton'),
		},
	);

	if (!confirmed) {
		return;
	}

	try {
		await environmentsStore.deleteVariable(data);
		allVariables.value = allVariables.value.filter((variable) => variable.id !== data.id);
	} catch (error) {
		showError(error, i18n.baseText('variables.errors.delete'));
	}
}

function goToUpgrade() {
	window.open(upgradeLinkUrl.value, '_blank');
}

function displayName(resource: EnvironmentVariable) {
	return resource.key;
}
</script>

<template>
	<ResourcesListLayout
		ref="layout"
		resource-key="variables"
		:disabled="!isFeatureEnabled"
		:resources="allVariables"
		:initialize="initialize"
		:shareable="false"
		:displayName="displayName"
		:sortOptions="['nameAsc', 'nameDesc']"
		:showFiltersDropdown="false"
		type="datatable"
		:type-props="{ columns: datatableColumns }"
		@click:add="addTemporaryVariable"
	>
		<template v-if="!isFeatureEnabled" #empty>
			<n8n-action-box
				data-test-id="empty-resources-list"
				emoji="ðŸ‘‹"
				:heading="$locale.baseText(contextBasedTranslationKeys.variables.unavailable.title)"
				:description="
					$locale.baseText(contextBasedTranslationKeys.variables.unavailable.description)
				"
				:buttonText="$locale.baseText(contextBasedTranslationKeys.variables.unavailable.button)"
				buttonType="secondary"
				@click="goToUpgrade"
			/>
		</template>
		<template #default="{ data }">
			<VariablesRow
				:key="data.id"
				:editing="editMode[data.id]"
				:data="data"
				@save="saveVariable"
				@edit="toggleEditingVariable"
				@cancel="toggleEditingVariable"
				@delete="deleteVariable"
			/>
		</template>
	</ResourcesListLayout>
</template>

<style lang="scss" module>
.type-input {
	--max-width: 265px;
}

.sidebarContainer ul {
	padding: 0 !important;
}
</style>
