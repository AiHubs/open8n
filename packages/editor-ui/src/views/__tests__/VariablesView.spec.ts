import { afterAll, beforeAll } from 'vitest';
import { setActivePinia, createPinia } from 'pinia';
import { setupServer } from '@/__tests__/server';
import { render } from '@testing-library/vue';
import VariablesView from '@/views/VariablesView.vue';
import { useSettingsStore } from '@/stores';

describe('store', () => {
	let server: ReturnType<typeof setupServer>;
	const seedRecordsCount = 3;

	beforeAll(() => {
		server = setupServer();
	});

	beforeEach(async () => {
		setActivePinia(createPinia());
		await useSettingsStore().getSettings();
	});

	afterAll(() => {
		server.shutdown();
	});

	it('should render empty state', () => {
		const wrapper = render(VariablesView);

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should render variable entries', () => {
		server.createList('variable', seedRecordsCount);

		const wrapper = render(VariablesView);

		expect(wrapper.html()).toMatchSnapshot();
	});
});
