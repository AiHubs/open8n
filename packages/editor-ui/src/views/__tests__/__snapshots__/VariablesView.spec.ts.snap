// Vitest Snapshot v1

exports[`store > should render empty state 1`] = `
"<div class=\\"wrapper\\">
  <div class=\\"container\\">
    <aside class=\\"aside\\">
      <div class=\\"heading-wrapper mb-xs\\"><span class=\\"n8n-heading size-2xlarge regular\\"> Variables </span></div>
      <div class=\\"mt-xs mb-l\\"><button aria-disabled=\\"false\\" aria-busy=\\"false\\" aria-live=\\"polite\\" class=\\"button button primary large block\\" data-test-id=\\"resources-list-add\\">
          <!----><span> Add Variable </span></button></div>
      <!---->
    </aside>
    <main class=\\"content\\">
      <div>
        <div class=\\"n8n-loading n8n-loading-custom header-loading mb-l\\">
          <div class=\\"el-skeleton is-animated\\">
            <div class=\\"custom\\">
              <div class=\\"el-skeleton__item el-skeleton__custom\\">
                <!---->
              </div>
            </div>
          </div>
        </div>
        <div class=\\"n8n-loading n8n-loading-custom card-loading mb-2xs\\">
          <div class=\\"el-skeleton is-animated\\">
            <div class=\\"custom\\">
              <div class=\\"el-skeleton__item el-skeleton__custom\\">
                <!---->
              </div>
            </div>
          </div>
        </div>
        <div class=\\"n8n-loading n8n-loading-custom card-loading\\">
          <div class=\\"el-skeleton is-animated\\">
            <div class=\\"custom\\">
              <div class=\\"el-skeleton__item el-skeleton__custom\\">
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>"
`;

exports[`store > should render variable entries 1`] = `
"<div class=\\"wrapper\\">
  <div class=\\"container\\">
    <aside class=\\"aside\\">
      <div class=\\"heading-wrapper mb-xs\\"><span class=\\"n8n-heading size-2xlarge regular\\"> Variables </span></div>
      <div class=\\"mt-xs mb-l\\"><button aria-disabled=\\"false\\" aria-busy=\\"false\\" aria-live=\\"polite\\" class=\\"button button primary large block\\" data-test-id=\\"resources-list-add\\">
          <!----><span> Add Variable </span></button></div>
      <!---->
    </aside>
    <main class=\\"content\\">
      <div>
        <div class=\\"n8n-loading n8n-loading-custom header-loading mb-l\\">
          <div class=\\"el-skeleton is-animated\\">
            <div class=\\"custom\\">
              <div class=\\"el-skeleton__item el-skeleton__custom\\">
                <!---->
              </div>
            </div>
          </div>
        </div>
        <div class=\\"n8n-loading n8n-loading-custom card-loading mb-2xs\\">
          <div class=\\"el-skeleton is-animated\\">
            <div class=\\"custom\\">
              <div class=\\"el-skeleton__item el-skeleton__custom\\">
                <!---->
              </div>
            </div>
          </div>
        </div>
        <div class=\\"n8n-loading n8n-loading-custom card-loading\\">
          <div class=\\"el-skeleton is-animated\\">
            <div class=\\"custom\\">
              <div class=\\"el-skeleton__item el-skeleton__custom\\">
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>"
`;
